jsonrpc/filter_manager.go

```diff
@@ -341,8 +366,17 @@ func (f *FilterManager) GetFilterChanges(id string) (string, error) {
 }
 
 func (f *FilterManager) Uninstall(id string) bool {
-	f.lock.Lock()
+	f.filtersLock.Lock()
+
+	removed := f.removeFilterByID(id)
+
+	f.filtersLock.Unlock()
 
+	return removed
+}
+
+// removeFilterByID removes a filter with given ID, unsafe agarint race condition
```

Comment:
Typo here, `agarint`

Reply:
Thank you for your kind alert. Fixed at dcb2346b957ab539d3e52daf2a313d1fec859913